### ISSUE TITLE
ltr329_simpletest.ino not working

### ISSUE THREAD

**Issue by JimKruk** [adafruit/Adafruit_LTR329_LTR303#1]
I was using this sensor with a UNO board with the adafruit ltr329 library v1.0.0 using the code sample ltr329_simpletest.ino and the readings do not seem valid.  The other example ltr329_advanced_test.ino does work.  I posted a q on the adafruit forums and the person who replied thinks there is a bug in the library with a missing & char.  

this is the link to the q I opened on adafruit, the person who replied suggested posting here since if appears to be a code bug. 
https://forums.adafruit.com/viewtopic.php?t=194850

thanks

**Comment by caternuson** [comment]
The library function being called is passing by reference:
https://github.com/adafruit/Adafruit_LTR329_LTR303/blob/f72c8b792857b0fc5f39a308f459cbf0a9e0bb64/Adafruit_LTR329_LTR303.cpp#L231
so the library code syntax is OK - it's not missing `&`s.

It's more likely the determination of `visible` requires more logic than the simple math being done:
https://github.com/adafruit/Adafruit_LTR329_LTR303/blob/f72c8b792857b0fc5f39a308f459cbf0a9e0bb64/Adafruit_LTR329_LTR303.cpp#L252

This same behavior can be seen with the [ltr329_advanced_test](https://github.com/adafruit/Adafruit_LTR329_LTR303/tree/main/examples/ltr329_advanced_test) example. That example does not output `visible`, just the two main channel readings. Running that example with a finger over the sensor produces values like this:
```
Visible + IR: 20		Infrared: 35
Visible + IR: 20		Infrared: 35
Visible + IR: 20		Infrared: 35
Visible + IR: 21		Infrared: 35
Visible + IR: 22		Infrared: 37
```

Note the infrared value is higher than the visible+infrared value. So the math for visible will result in a negative number. You're seeing the values in the 65000 range since the return value is unsigned (`uint16_t`).

**Comment by JimKruk** [comment]
Thanks for the quick reply.  I am not an engineer but how can you have a
negative light value ?  seems like the library needs to be updated to
return a valid visible level ? Or change the tuning setup values to provide
a valid value back ?

btw - I did try to change the code to use the & as a user suggested but it
would not compile :(

thanks

**Comment by caternuson** [comment]
You can't have a negative light level. The value was erroneous. The solution, for now, is going to be to remove the `readVisible()` method.

The general issue is that CH0, the nominal "visible" channel, has some sensitivity in the infrared range. CH1, the nominal "infrared" channel, can potentially be used to correct for that. However, exactly how that should be done math-wise is currently not known. But it's not as simple as what was being done, which was "visible" = CH0 - CH1.

There is some related discussion others have pointed to here:
https://forum.digikey.com/t/ltr-329-lux-calculation/15776
but that is specific to lux, which is related but different.

**Comment by JimKruk** [comment]
Ok, thanks for the clarification. Not sure if you are affiliated with
adafruit but maybe they should remove the simple test program from the
examples or add a disclaimer it may return negative values.  I think it
will cause confusion with basic users like myself. Thanks

**Comment by caternuson** [comment]
Yep. That's in the PR:
https://github.com/adafruit/Adafruit_LTR329_LTR303/pull/2

**Comment by caternuson** [comment]
The PR has been merged and released with 2.0.0:
https://github.com/adafruit/Adafruit_LTR329_LTR303/releases/tag/2.0.0

Not really a fix, since the erroneous method was simply removed.

For "visible" light, suggest just using the CH0 value. The fact it contains some amount of infrared light as well should hopefully not matter for most basic light sensing applications.

**Comment by JimKruk** [comment]
Great, thanks!
